prosody-plugins/token/util.lib.lua

```diff
@@ -38,19 +29,12 @@ local function _verify_token(token, appId, appSecret, roomName, disableRoomNameC
 	if roomClaim == nil and disableRoomNameConstraints ~= true then
 		return nil, "'room' claim is missing";
 	end
-	if roomName ~= nil and roomName ~= roomClaim and disableRoomNameConstraints ~= true then
-		return nil, "Invalid room name('room' claim)";
-	end
 
 	return true;
 end
 
-function _M.verify_token(token, appId, appSecret, roomName, disableRoomNameConstraints)
-	return _verify_token(token, appId, appSecret, roomName, disableRoomNameConstraints);
-end
-
-function _M.get_room_name(token, appSecret)
```

Comment:
It wasn't used; the room name was extracted from the token and then passed in to get_room_name which extracted it from the token and then compared them, I think. I'll go back and double check. The token was also being decoded and its signature verified on the call to get_room_name and the call to verify_token, which was duplicating some effort.